zerver/lib/import_realm.py

```diff
@@ -662,6 +664,16 @@ def import_uploads(import_dir: Path, processing_avatars: bool=False,
                         user_profile.id))
                     # Delete the record of the avatar to avoid 404s.
                     do_change_avatar_fields(user_profile, UserProfile.AVATAR_FROM_GRAVATAR)
+            return 0
+
+        if threads == 1:
+            for record in records:
+                process_avatars(record)
+        else:
+            connection.close()
+            output = []
+            for (status, job) in run_parallel(process_avatars, records, threads):
+                output.append(job)
```

Comment:
Did you test this using the actual S3 backend?

Reply:
@timabbott hmm. Actually, I moved the import to inside the try block since I thought the import was used only there. There was no other reason for that. It was probably a bad change since now 'upload_backend' is imported everytime the process is run. I will just move it back to the original position. I have not tested in s3 yet. I tried the import only using the default settings. Will try it in S3.